R/summary_functions.R

```diff
@@ -4,8 +4,9 @@
 #' @export
 percent_coverage <- function(coverage_result){
   coverage_data <- as.data.frame(coverage_result)
-  num_lines <- nrow(coverage_data)
-  num_not_zero <- sum(coverage_data$value != 0) # how many lines have a value that is not zero?
+  num_lines <- length(unique(coverage_data$first_line))
```

Comment:
Not sure about the motivation for these changes, they seem to make the intent less clear to me.

This function should probably just be made into two S3 functions
```r
percent_coverage <- function(coverage_result) {
  UseMethod("percent_coverage")
}

percent_coverage.coverage <- function(coverage_result) {
  sum(x > 0)  / length(x)
}

percent_coverage.data.frame <- function(coverage_result) {
  sum(coverage_result$value > 0) / NROW(coverage_result)
}
```